include/ensmallen_bits/moead/moead_impl.hpp

```diff
@@ -238,28 +259,30 @@ Optimize(std::tuple<ArbitraryFunctionType...>& objectives,
       Mutate(candidate, 1.0 / static_cast<double>(numVariables),
           castedLowerBound, castedUpperBound);
 
-      arma::Col<ElemType> candidateFitness(numObjectives);
+          ColType candidateFitness(numObjectives);
       //! Creating temp vectors to pass to EvaluateObjectives.
       std::vector<BaseMatType> candidateContainer { candidate };
-      std::vector<arma::Col<ElemType>> fitnessContainer { candidateFitness };
+      std::vector<ColType> fitnessContainer { candidateFitness };
       EvaluateObjectives(candidateContainer, objectives, fitnessContainer);
       candidateFitness = std::move(fitnessContainer[0]);
       //! Flush out the dummy containers.
       fitnessContainer.clear();
       candidateContainer.clear();
 
       // 2.4 Update of ideal point.
-      idealPoint = arma::min(idealPoint, candidateFitness);
+      idealPoint = min(idealPoint, candidateFitness);
 
       // 2.5 Update of the population.
       size_t replaceCounter = 0;
       const size_t sampleSize = sampleNeighbor ? neighborSize : populationSize;
 
-      const arma::uvec idxShuffle = arma::shuffle(
-          arma::linspace<arma::uvec>(0, sampleSize - 1, sampleSize));
+      const UVecType idxShuffle = shuffle(
+          linspace<UVecType>(0, sampleSize - 1, sampleSize));
```

Comment:
I think this might make more sense to always keep as an `arma::uvec` since we are always iterating over the individual elements on the CPU.